scaaml/aes_forward.py

```diff
@@ -109,4 +112,6 @@ def get_attack_point(cls, name: str, **kwargs: bytearray) -> bytearray:
         function = getattr(cls, name, None)
         if not function or not callable(function):
             raise ValueError(f"{name} is not a defined attack point")
-        return function(**kwargs)
+        function = cast(Callable[[bytearray, bytearray], bytearray], function)
```

Comment:
NIT (by all means feel free to ignore / leave that for me or another PR) The parameter `name` has type `str` which is nice since as soon as we add another method we do not need to change this method. But it also means we are losing typechecking especially when there would be more methods. WDYT `assert type(function) is Callable[[bytearray, bytearray], bytearray]`?

Reply:
We don't really loose anything. Type checking in Python is static and by default nothing happens at runtime.
The cast operation is only to make the typechecker happy.
But we can add runtime type checking too if you want.
I'd be more worried about the input generators though. I started typing them and it still uses Any because I wasn't sure about the types we want.

Reply:
Agree we are not losing anything, we could gain here a little bit, but I don't think that is necessary (very unlikely and the error message would be similar).

Ad input generators definitely planned to consolidate bytearrays and np arrays when moving to those.